ml_tools/eolearn/ml_tools/TDigestTask.py

```diff
@@ -0,0 +1,168 @@
+
+                else:
+                    eopatch[out_feature] = np.empty(shape[[0, -1]], dtype=object)
+                    for k in range(shape[-1]):
+                        eopatch[out_feature][time_, k] = td.TDigest()
+                        eopatch[out_feature][time_, k].batch_update(eopatch[in_feature][time_, ..., k].flatten())
+
+        elif self.mode == "monthly":
+            midx = []
+            for month_ in range(12):
+                midx.append(np.array([timestamp.month == month_ + 1 for timestamp in eopatch["timestamp"]]))
+
+            for in_feature, out_feature in zip(self.in_feature, self.out_feature):
+                shape = np.array(eopatch[in_feature].shape)
+                if self.pixelwise:
+                    eopatch[out_feature] = np.empty([12, *shape[1:]], dtype=object)
+                    for month_, i, j, k in product(range(12), range(shape[1]), range(shape[2]), range(shape[3])):
+                        eopatch[out_feature][month_, i, j, k] = td.TDigest()
+                        eopatch[out_feature][month_, i, j, k].batch_update(
+                            eopatch[in_feature][midx[month_], i, j, k].flatten()
+                        )
+
+                else:
+                    eopatch[out_feature] = np.empty([12, shape[-1]], dtype=object)
+                    for month_, k in product(range(12), range(shape[-1])):
+                        eopatch[out_feature][month_, k] = td.TDigest()
+                        eopatch[out_feature][month_, k].batch_update(
+                            eopatch[in_feature][midx[month_], ..., k].flatten()
+                        )
+
+        elif self.mode == "total":
+            for in_feature, out_feature in zip(self.in_feature, self.out_feature):
+                eopatch[out_feature] = td.TDigest()
+                eopatch[out_feature].batch_update(eopatch[in_feature].flatten())
+
+        else:
+            raise RuntimeError(f"TDigestTask: mode {self.mode} not implemented!")
+
+        return eopatch
+
+
+def _get_allowed_ftypes(featuretype, ifin=True, mode="standard", pixelwise=False):
+    if mode == "standard":
+        allowed_in_types = [
+            FeatureType.DATA,
+            FeatureType.DATA_TIMELESS,
+            FeatureType.MASK,
+            FeatureType.MASK_TIMELESS,
+        ]
+        allowed_out_types = [FeatureType.DATA_TIMELESS] if pixelwise else [FeatureType.SCALAR_TIMELESS]
+
+    elif mode == "timewise" or mode == "monthly":
+        allowed_in_types = [FeatureType.DATA, FeatureType.MASK]
+        allowed_out_types = [FeatureType.DATA] if pixelwise else [FeatureType.SCALAR]
+
+    elif mode == "total":
+        allowed_in_types = ...
+        allowed_out_types = [FeatureType.SCALAR_TIMELESS]
+
+    if ifin:
+        return True if allowed_in_types is Ellipsis else featuretype in allowed_in_types
+    else:
+        return True if allowed_out_types is Ellipsis else featuretype in allowed_out_types
```

Comment:
```suggestion
ModeTypes = Literal["standard", "timewise", "monthly", "total"]


def _is_input_ftype(feature_type: FeatureType, mode: ModeTypes) -> bool:
    if mode == "standard":
        return feature_type.is_image()
    if mode in ("timewise", "monthly"):
        return feature_type in [FeatureType.DATA, FeatureType.MASK]
    return True


def _is_output_ftype(feature_type: FeatureType, mode: ModeTypes, pixelwise: bool) -> bool:
    if mode == "standard":
        return feature_type == (FeatureType.DATA_TIMELESS if pixelwise else FeatureType.SCALAR_TIMELESS)

    if mode in ("timewise", "monthly"):
        return feature_type == (FeatureType.DATA if pixelwise else FeatureType.SCALAR)

    return feature_type == FeatureType.SCALAR_TIMELESS
```

This can be made simpler by separating the input and output type function. I also introduced a custom type, which you can also use in the `__init__` (the docs will inline the actual `Literal` type, so this is just for our convenience)